index.d.ts

```diff
@@ -0,0 +1,15 @@
+declare module "messageformat" {
+    class MessageFormat {
+        constructor(message: { [pluralFuncs: string]: Function });
+        constructor(message: string[]);
+        constructor(message: string);
+        addFormatters: (format: {}) => MessageFormat;
```

Comment:
I think this could be:
```typescript
type Formatter = (val: any, lc: string, arg?: string) => string;
addFormatters: (format: { [name: string]: Formatter }) => MessageFormat;
```